bot/cogs/events.py

```diff
@@ -0,0 +1,152 @@
+# coding=utf-8
+import logging
+
+from discord import Embed, Member
+from discord.ext.commands import (
+    AutoShardedBot, BadArgument, BotMissingPermissions,
+    CommandError, CommandInvokeError, Context,
+    NoPrivateMessage, UserInputError
+)
+
+from bot.constants import (
+    ADMIN_ROLE, DEVLOG_CHANNEL, DEVOPS_ROLE, MODERATOR_ROLE, OWNER_ROLE, PYTHON_GUILD, SITE_API_KEY, SITE_API_USER_URL
+)
+
+log = logging.getLogger(__name__)
+
+
+class Events:
+    """
+    No commands, just event handlers
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+
+    async def send_updated_users(self, *users):
+        try:
+            response = await self.bot.http_session.post(
+                url=SITE_API_USER_URL,
+                json=list(users),
+                headers={"X-API-Key": SITE_API_KEY}
+            )
+
+            return await response.json()
+        except Exception as e:
+            log.error(f"Failed to send role updates: {e}")
+            return {}
+
+    async def on_command_error(self, ctx: Context, e: CommandError):
+        command = ctx.command
+        parent = None
+
+        if command is not None:
+            parent = command.parent
+
+        if parent and command:
+            help_command = (self.bot.get_command("help"), parent.name, command.name)
+        elif command:
+            help_command = (self.bot.get_command("help"), command.name)
+        else:
+            help_command = (self.bot.get_command("help"),)
+
+        if isinstance(e, BadArgument):
+            await ctx.send(f"Bad argument: {e}\n")
+            await ctx.invoke(*help_command)
+        elif isinstance(e, UserInputError):
+            await ctx.invoke(*help_command)
+        elif isinstance(e, NoPrivateMessage):
+            await ctx.send("Sorry, this command can't be used in a private message!")
+        elif isinstance(e, BotMissingPermissions):
+            await ctx.send(
+                f"Sorry, it looks like I don't have the permissions I need to do that.\n\n"
+                f"Here's what I'm missing: **{e.missing_perms}**"
+            )
+        elif isinstance(e, CommandInvokeError):
+            await ctx.send(
+                f"Sorry, an unexpected error occurred. Please let us know!\n\n```{e}```"
+            )
+            raise e.original
+        log.error(f"COMMAND ERROR: '{e}'")
```

Comment:
This isn't logged if a non-expected CommandInvokeError happens. You could either wrap the ifs in a try - finally block, however just logging the error before you raise it would be fine.